### PR TITLE
gcc7 compilation warning(s) fix(es) in DataFormats/L1THGCal

### PR DESCRIPTION
**Pull request by mrodozov**
Fix for compilation warning(s) when compiling with gcc7 and more flags listed here:
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc700/CMSSW_9_3_X_2017-07-24-2300/L1Trigger/L1THGCal

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by kpedro88** [comment]
+1

**Comment by davidlange6** [comment]
merge

File: DataFormats/L1THGCal/interface/HGCFETriggerDigi.h [hunk 1]
```diff
@@ -43,7 +43,9 @@ namespace l1t {
     typedef std::vector<bool> data_payload;
     typedef uint32_t key_type; 
 
-    HGCFETriggerDigi() : codec_((unsigned char)0xffff) {}
+    HGCFETriggerDigi() : codec_((unsigned char)0xffff) {
+      detid_ = 0;
```
Comment (Dr15Jones) [inline DataFormats/L1THGCal/interface/HGCFETriggerDigi.h]: The better way is to do it as part of the initializer list
```
HGCFETriggerDigi() : codec_((unsigned char)0xffff), detid_(0) {}

**Pull request merged**